whylog/teacher/__init__.py

```diff
@@ -159,6 +140,20 @@ def set_causes_relation(self, relation):
         """
         pass
 
+    def make_group(self, line_id, span):
+        """
+        Improves text pattern by adding group corresponding to param in text.
+        Removes (or maybe updates) constraints related to groups in line with line_id
+        """
+        pass
+
+    def remove_group(self, pattern_group):
```

Comment:
We should ask Client if we need it and if yes, we can create proper mechanism in Front. I suggest not to remove it yet.